paddle/fluid/pybind/CMakeLists.txt

```diff
@@ -1,9 +1,16 @@
 if(WITH_PYTHON)
```

Comment:
What's the purpose of updating this `pybind/CMakeLists.txt`?

Reply:
The reason of updating pybind/CMakeLists.txt is that shared library which refers any HIP kernel must be linked with hipcc rather than gcc.